sklearn/utils/validation.py

```diff
@@ -2706,30 +2706,27 @@ def _to_object_array(sequence):
 
 
 def _check_feature_names(estimator, X, *, reset):
-    """Set or check the `feature_names_in_` attribute of an estimator.
+    """Check or set the `feature_names_in_` attribute for an estimator.
 
-    .. versionadded:: 1.0
+       This is typically used when `validate=False` during fitting, to still
+       record the feature names.
```

Comment:
It has nothing to do with ``validate``. It always happens unless in very specific cases. and ``validate`` is not a common parameter of estimators, only a few of them have it). So I think this sentence is misleading and I would not include it.
```suggestion
```